### PR TITLE
#7255: Update content script readiness checks to drop mentions of "manual injection"

### PR DESCRIPTION
**Pull request by fregante**
## What does this PR do?

- Closes https://github.com/pixiebrix/pixiebrix-extension/issues/7255
- Follows https://github.com/pixiebrix/pixiebrix-extension/pull/8120

## Checklist

- [x] Add tests and/or storybook stories
- [x] Designate a primary reviewer: @grahamlangford

### CODE REVIEW

File: src/contentScript/ready.ts [hunk 1]
```diff
@@ -33,79 +32,85 @@
...
+    isRemoteProcedureCallRequest(message) &&
+    message.type === CONTENT_SCRIPT_READINESS_CHECK
+  ) {
+    // Do not return an unpromised `true` because `webextension-polyfill` handles it differently
+    return Promise.resolve(true);
+  }
 }
 
+// TODO: Move out of contentScript/ready.ts, it's unrelated logic
 let reloadOnNextNavigate = false;
```
Comment (fregante) [inline src/contentScript/ready.ts:114]: This belongs to `contentScript/lifecycle.ts`, but moving it there causes plenty of `strictNullChecks` issues

File: src/contentScript/ready.ts [hunk 2]
```diff
@@ -33,79 +32,85 @@
...
+      );
+    }
+
+    // Notify `waitForContentScript` listeners, if any
+    void browser.runtime.sendMessage({
+      type: CONTENT_SCRIPT_READY_NOTIFICATION,
+    });
+
+    // Respond to any successive `waitForContentScript` pings
+    browser.runtime.onMessage.addListener(respondToReadinessChecks);
```
Comment (fregante) [inline src/contentScript/ready.ts:77]: This listener is only added when the script is ready, and it only responds `true`

File: src/contentScript/ready.ts [hunk 3]
```diff
@@ -33,79 +32,79 @@
...
+    { frameId: target.frameId },
+  )) as true | undefined;
+  return response ?? false;
 }
 
-/**
- * Fetches the URL and content script state from tab/frame
- * @throws Error if background page doesn't have permission to access the tab
- */
-export async function getTargetState(target: Target): Promise<TargetState> {
```
Comment (fregante) [inline src/contentScript/ready.ts:87]: We really only used the `ready` state, so I changed `getReadyState` (which used scripting) to this message, which exclusively responds `true` or `undefined` (i.e. no content script or no "ready" listener attached yet)

File: src/testUtils/extendedExpectations.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+async function trackSettleTime(promise: Promise<unknown>): Promise<number> {
+  const startTime = performance.now();
+  try {
+    await promise;
+  } catch {}
+
+  return performance.now() - startTime;
+}
+
+expect.extend({
```
Comment (fregante) [inline src/testUtils/extendedExpectations.ts:45]: More custom expectations should use this pattern

https://github.com/pixiebrix/pixiebrix-extension/blob/34ca06ed91a4f07e404c32d3d2350defff558306/src/testUtils/testItRenders.tsx#L24

Comment (grahamlangford) [reply src/testUtils/extendedExpectations.ts:45]: It's been a low priority goal of mine to get rid of `testItRenders` and similar.

File: src/testUtils/extendedExpectations.ts [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  // Extracted and adapted from https://github.com/pixiebrix/webext-messenger/blob/22eeba8b5b2efe3ecb6beb7a8c493f260c9499fb/source/test/helpers.ts#L40
+  async toFullfillWithinMilliseconds(
```
Comment (grahamlangford) [inline src/testUtils/extendedExpectations.ts:47]: I think we should comment that this only asserts on the amount of time that passes from when the expectation is asserted, not when the promise is actually invoked. (Could leak to confusing behavior or false confidence if we make other assertions first).

Comment (fregante) [reply src/testUtils/extendedExpectations.ts:47]: Yeah promise timing assertions are a bit tricky. In the messenger repo I actually had a separate promise tracking function for this reason, but it doesn't fit greatly with the `expect()` call pattern

https://github.com/pixiebrix/webext-messenger/blob/20f7773b73c666412db579d496bd0156214eb2c9/source/test/background/api.test.ts#L103-L111

I'll add a comment

**Pull request merged**